### PR TITLE
Loading factions from JSON files

### LINKED ISSUE #11: Loading unit types, building types and factions from files
**A short description of the task**
Unit types, building types and factions should be defined in files that are then loaded into the game.
This creates flexibility to quickly alter these data.

**Describe the solution you'd like**
Use a format like YAML to define the different types.

- Factions are collections of buildings that are available to a faction.
- Buildings contain sprite, stats and which unit it spawns.
- Units contain sprite and stats.

**Describe alternatives you've considered**
It could also be implemented using JSON or another format, but I think YAML would be easy to read and write.

**Additional context**
Rust has several crates ready to serialise/deserialise data from different formats and probably also works with YAML or has a companion crate that does that.

### PR DESCRIPTION
**Pull request by phillipphoenix**
It is now possible to add factions via JSON files in the factions folder. Added a human faction with a building and a unit. Also added a markdown file explaining the format somewhat.

Closes #11

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -10,6 +10,9 @@ bevy = "0.13.0"
 bevy-inspector-egui = "0.23.4"
 bevy_rapier2d = { version = "0.25.0"}
 bevy_ecs_ldtk = { git = "https://github.com/theshortcut/bevy_ecs_ldtk/", branch = "bevy-0.13" }
+serde = { version = "1.0.197", features = ["derive"] }
+thiserror = "1.0.58"
```
Comment (Blavion) [inline Cargo.toml:14]: supportere det her custom errors? 
Som I vil den altid smide en `std::error:Error` ??

File: src/load_factions.rs [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    fn build(&self, app: &mut App) {
+        app.register_asset_loader::<FactionAssetLoader>(FactionAssetLoader)
+            .init_asset::<FactionAsset>()
+            .insert_resource(Factions(vec![]))
+            .add_systems(Startup, load_factions)
+            .add_systems(
+                Update,
+                (
+                    setup_factions_resource.after(load_factions),
+                    display_content.after(setup_factions_resource), // TODO: Remove display_content, when actually using the factions resource.
```
Comment (Blavion) [inline src/load_factions.rs:26]: Are we accepting `TODO` as comments, or should we make them as issues for tracking?

**Review by Blavion — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (phillipphoenix) [reply Cargo.toml:14]: Den kommer med noget macro mm. der gør det nemmere at arbejde med errors. Det kan ses ved `FactionLoaderError` under load_factions.rs

File: src/load_factions.rs (hunk 2 above)
Comment (phillipphoenix) [reply src/load_factions.rs]: Der er oprettet opgave på den.

**Pull request merged**